### PR TITLE
OV-105: add avatars endpoint

### PR DESCRIPTION
**Pull request by WilsonBravo**
### Avatar Properties Relationship Example

| Character | Style    | Gesture |
|-----------|----------|------------|
| Harry     | business | 123        |
| Harry     | business | calm-down  |
| Harry     | business | come-on    |
| Harry     | business | five-star-reviews |
| Harry     | business | good       |
| Harry     | business | hello      |
| Harry     | business | introduce  |
| Harry     | business | invite     |
| Harry     | business | thanks     |
| Harry     | business | welcome    |
| Harry     | casual   | 123        |
| Harry     | casual   | come-on    |
| Harry     | casual   | five-star-reviews |
| Harry     | casual   | gong-xi-fa-cai |
| Harry     | casual   | good       |
| Harry     | casual   | happy-new-year |
| Harry     | casual   | hello      |
| Harry     | casual   | please     |
| Harry     | casual   | welcome    |
| Harry     | youthful | 123        |
| Harry     | youthful | come-on    |
| Harry     | youthful | down       |
| Harry     | youthful | five-star  |
| Harry     | youthful | good       |
| Harry     | youthful | hello      |
| Harry     | youthful | invite     |
| Harry     | youthful | show-right-up-down |
| Harry     | youthful | welcome    |

### Avatars
```json
{
    "items": [
        {
            "id": "b58f9707-e4a4-49fc-8270-5984e70deb70",
            "name": "harry",
            "voice": "en-US-AndrewNeural",
            "voiceUrl": "https://speechstudioprodpublicsa.blob.core.windows.net/ttsvoice/Masterpieces/en-US-Andrew-General-Audio.wav",
            "styles": [
                {
                    "imgUrl": "https://speech.microsoft.com/assets/avatar/harry/harry-business-thumbnail.png",
                    "style": "business",
                    "gestures": [
                        "123",
                        "calm-down",
                        "come-on",
                        "five-star-reviews",
                        "good",
                        "hello",
                        "introduce",
                        "invite",
                        "thanks",
                        "welcome"
                    ]
                },
                {
                    "imgUrl": "https://speech.microsoft.com/assets/avatar/harry/harry-casual-thumbnail.png",
                    "style": "casual",
                    "gestures": [
                        "123",
                        "come-on",
                        "five-star-reviews",
                        "gong-xi-fa-cai",
                        "good",
                        "happy-new-year",
                        "hello",
                        "please",
                        "welcome"
                    ]
                },
                {
                    "imgUrl": "https://speech.microsoft.com/assets/avatar/harry/harry-youthful-thumbnail.png",
                    "style": "youthful",
                    "gestures": [
                        "123",
                        "come-on",
                        "down",
                        "five-star",
                        "good",
                        "hello",
                        "invite",
                        "show-right-up-down",
                        "welcome"
                    ]
                }
            ]
        },
        {
            "id": "e531c2c9-e06b-40f8-9fda-7b4168f79640",
            "name": "jeff",
            "voice": "en-US-BrandonNeural",
            "voiceUrl": "https://speechstudioprodpublicsa.blob.core.windows.net/ttsvoice/Masterpieces/en-US-Brandon-General-Audio.wav",
            "styles": [
                {
                    "imgUrl": "https://speech.microsoft.com/assets/avatar/jeff/jeff-business-thumbnail-bg.png",
                    "style": "business",
                    "gestures": [
                        "123",
                        "come-on",
                        "five-star-reviews",
                        "hands-up",
                        "here",
                        "meddle",
                        "please2",
                        "show",
                        "silence",
                        "thanks"
                    ]
                },
                {
                    "imgUrl": "https://speech.microsoft.com/assets/avatar/jeff/jeff-formal-thumbnail-bg.png",
                    "style": "formal",
                    "gestures": [
                        "123",
                        "come-on",
                        "five-star-reviews",
                        "lift",
                        "please",
                        "silence",
                        "thanks",
                        "very-good"
                    ]
                }
            ]
        },
    ]
}
```

### Avatar by ID
```json
{
    "id": "b58f9707-e4a4-49fc-8270-5984e70deb70",
    "name": "harry",
    "voice": "en-US-AndrewNeural",
    "voiceUrl": "https://speechstudioprodpublicsa.blob.core.windows.net/ttsvoice/Masterpieces/en-US-Andrew-General-Audio.wav",
    "styles": [
        {
            "imgUrl": "https://speech.microsoft.com/assets/avatar/harry/harry-business-thumbnail.png",
            "style": "business",
            "gestures": [
                "123",
                "calm-down",
                "come-on",
                "five-star-reviews",
                "good",
                "hello",
                "introduce",
                "invite",
                "thanks",
                "welcome"
            ]
        },
        {
            "imgUrl": "https://speech.microsoft.com/assets/avatar/harry/harry-casual-thumbnail.png",
            "style": "casual",
            "gestures": [
                "123",
                "come-on",
                "five-star-reviews",
                "gong-xi-fa-cai",
                "good",
                "happy-new-year",
                "hello",
                "please",
                "welcome"
            ]
        },
        {
            "imgUrl": "https://speech.microsoft.com/assets/avatar/harry/harry-youthful-thumbnail.png",
            "style": "youthful",
            "gestures": [
                "123",
                "come-on",
                "down",
                "five-star",
                "good",
                "hello",
                "invite",
                "show-right-up-down",
                "welcome"
            ]
        }
    ]
}
```

### CODE REVIEW

File: backend/src/migrations/20240830182023_add_avatars_table.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const TABLE_NAME = 'avatars';
+
+const ColumnName = {
+    ID: 'id',
+    NAME: 'name',
+    VOICE_URL: 'voice_url',
+    VOICE: 'voice',
+    CREATED_AT: 'created_at',
+    UPDATED_AT: 'updated_at',
+};
```
Comment (nikita-remeslov) [inline backend/src/migrations/20240830182023_add_avatars_table.ts:12]:
```suggestion
} as const;
```

File: backend/src/migrations/20240830183716_add_styles_table.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'styles';
+
+const ColumnName = {
+    ID: 'id',
+    STYLE: 'style',
+};
```
Comment (nikita-remeslov) [inline backend/src/migrations/20240830183716_add_styles_table.ts:8]:
```suggestion
} as const;
```

File: backend/src/migrations/20240830185525_add_gestures_table.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'gestures';
+
+const ColumnName = {
+    ID: 'id',
+    GESTURE: 'gesture',
+};
```
Comment (nikita-remeslov) [inline backend/src/migrations/20240830185525_add_gestures_table.ts:8]:
```suggestion
} as const;
```

File: backend/src/migrations/20240830190054_add_avatars_styles_table.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'avatars_styles';
+
+const ColumnName = {
+    ID: 'id',
+    AVATAR_ID: 'avatar_id',
+    STYLE_ID: 'style_id',
+    IMG_URL: 'img_url',
+};
```
Comment (nikita-remeslov) [inline backend/src/migrations/20240830190054_add_avatars_styles_table.ts:10]:
```suggestion
} as const;
```

File: backend/src/migrations/20240830191713_add_avatars_styles_gestures_table.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'avatars_styles_gestures';
+
+const ColumnName = {
+    AVATAR_STYLE_ID: 'avatar_style_id',
+    GESTURE_ID: 'gesture_id',
+};
```
Comment (nikita-remeslov) [inline backend/src/migrations/20240830191713_add_avatars_styles_gestures_table.ts:8]:
```suggestion
} as const;
```

File: backend/src/migrations/20240830202033_seed_styles_data.ts
Comment (nikita-remeslov) [inline backend/src/migrations/20240830202033_seed_styles_data.ts:1]: seeds !== migrations

File: backend/src/migrations/20240830202033_seed_styles_data.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'styles';
+
+async function up(knex: Knex): Promise<void> {
+    await knex(TABLE_NAME).insert([
+        { id: 'b58f9707-e4a4-49fc-8270-5984e70deb70', style: 'business' },
```
Comment (nikita-remeslov) [inline backend/src/migrations/20240830202033_seed_styles_data.ts:7]: your seed can fail if you hard code uuid

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Looks like we can just save config file at the backend. Because if Azure changes something you need to create additional migration to change everything

File: backend/src/migrations/20240830202033_seed_styles_data.ts (hunk 6 above)
Comment (WilsonBravo) [reply backend/src/migrations/20240830202033_seed_styles_data.ts:7]: Sorry if I didn't understand, but should I create a JSON file to store the avatar data and manage everything from there? This way, we can keep the configuration static and avoid needing database changes if Azure updates anything.

```
── /avatars
│   ├── /config
│   │   └── avatars.json
│   │
│   ├── /types
│   │
│   ├── /enums
│   │
│   ├── avatar.service.ts
│   │
│   ├── avatar.controller.ts 
│   │
│   └── avatars.ts
```

Comment (WilsonBravo) [reply backend/src/migrations/20240830202033_seed_styles_data.ts]: I made the changes

**Review by o-nedashkivska — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Pull request merged**